spec/inputs/imap_spec.rb

```diff
@@ -132,4 +134,27 @@
       insist { event.get("message") } == msg_text
     end
   end
-end
+
+  context "with attachments" do
+    it "should extract filenames" do
+      config = {"type" => "imap", "host" => "localhost",
+                "user" => "#{user}", "password" => "#{password}"}
+
+      input = LogStash::Inputs::IMAP.new config
+      input.register
+      event = input.parse_mail(subject)
+      insist { event.get("attachments") } == [{"filename"=>"some.html"}, {"filename"=>"image.png"}]
+    end
+
+    it "should extract the base64 content" do
+      config = {"type" => "imap", "host" => "localhost",
+        "user" => "#{user}", "password" => "#{password}",
+        "save_attachments" => true}
+
+      input = LogStash::Inputs::IMAP.new config
+      input.register
+      event = input.parse_mail(subject)
+      insist { event.get("attachments") } == [{"data"=>"PHA+YSBwYXJhZ3JhcGg8L3A+Cgo=\r\n", "filename"=>"some.html"}, {"data"=>"QkNE\r\n", "filename"=>"image.png"}]
```

Comment:
It would be preferable to call out to the `Base64` module on the original text to convert the text rather than use a pre-converted text string

Reply:
Done